test/e2e_clutz_dts_test.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// tslint:disable:no-unused-expression mocha .to.be.empty getters.
+
+import {expect} from 'chai';
+import * as ts from 'typescript';
+
+import * as tsickle from '../src/tsickle';
+
+import {compileWithTransfromer, createProgram, dtsCompilerOptions} from './test_support';
+
+describe('clutz dts', () => {
+  it('does nothing if there are no exports', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `const x = 1;`);
+
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    expect(dts).not.to.contain('namespace ಠ_ಠ.clutz');
+  });
+
+  it('produces a valid .d.ts', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `export const x = 1;`);
+
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    const program = createProgram(new Map([['input.d.ts', dts]]));
+    const diagnostics = ts.getPreEmitDiagnostics(program);
+
+    expect(diagnostics, tsickle.formatDiagnostics(diagnostics)).to.be.empty;
+  });
+
+  it('reexports things in the clutz namespace', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `
+      export const x = 1;
+      export function incr(n) { return x + 1; }`);
+
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    expect(dts).to.contain('namespace ಠ_ಠ.clutz');
+    expect(dts).to.contain('const module$contents$input_x: typeof x;');
+    expect(dts).to.contain('const module$contents$input_incr: typeof incr;');
```

Comment:
shouldn't tsickle emit something that can be directly imported in JS land, i.e. the whole external ambient module declaration with a `goog:` prefix? That seems a lot less brittle than relying on some Closure Compiler name mangling scheme to transport the type over.